src/actions/featureFlagsActions.js

```diff
@@ -18,18 +18,32 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
-import { getRemoteFeatureFlags } from 'services/featureFlags';
-import { SET_FEATURE_FLAGS } from 'constants/featureFlagsConstants';
+import get from 'lodash.get';
+
+import {
+  DEVELOPMENT_FEATURE_FLAGS,
+  INITIAL_FEATURE_FLAGS,
+  SET_FEATURE_FLAGS,
+} from 'constants/featureFlagsConstants';
 import { saveDbAction } from 'actions/dbActions';
-import type { GetState } from 'reducers/rootReducer';
+import { isProdEnv, isTest } from 'utils/environment';
+import type { Dispatch, GetState } from 'reducers/rootReducer';
+import type SDKWrapper from 'services/api';
 
-export const fetchFeatureFlagsAction = () => {
-  return async (dispatch: Function, getState: GetState) => {
-    const {
-      session: { data: { isOnline } },
-    } = getState();
-    if (!isOnline) return;
-    const featureFlags = await getRemoteFeatureFlags();
+export const loadFeatureFlagsAction = (userInfo?: any) => {
+  return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
+    if (!userInfo) {
+      const walletId = get(getState(), 'user.data.walletId');
+      userInfo = await api.userInfo(walletId);
```

Comment:
How about fall backing to the cached flags when the user is online?

Reply:
Currently we use `loadFeatureFlagsAction` in 3 cases, 2 of them (login and onboarding) passes `userInfo` that is retrieved few calls before fetching feature flags and 3rd case that doesn't pass `userInfo` actually needs latest flags to be fetched. I think caching won't provide any improvement.

Reply:
makes sense 👍